packages/fetchai/connections/p2p_noise/noise_node.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	//"strings"
+	"time"
+	"errors"
+	"github.com/perlin-network/noise"
+	"github.com/perlin-network/noise/kademlia"
+  	aea "./aea"
+)
+
+// check panics if err is not nil.
+func check(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+
+// An initial noise p2p node for AEA's fetchai/p2p-noise/0.1.0 connection
+func main() {
+	
+	// Create connection to aea
+	agent := aea.AeaApi{}
+	check(agent.Init())
+
+	// Create a new configured node.
+  	host, port := agent.Uri()
+  	key, err := noise.LoadKeysFromHex(agent.PrivateKey())
+	
+  	node, err := noise.NewNode(
+		noise.WithNodeBindHost(host),
+		noise.WithNodeBindPort(port),
+		noise.WithNodeAddress(""),
+    	noise.WithNodePrivateKey(key),
+	)
+	check(err)
+
+	// Release resources associated to node at the end of the program.
+	defer node.Close()
+
+	// Register Envelope message
+	node.RegisterMessage(aea.Envelope{}, aea.UnmarshalEnvelope)
+
+	// Register a message handler to the node.
+	node.Handle(func (ctx noise.HandlerContext) error {
+    	return handle(ctx, agent)
```

Comment:
(minor) > indentation

Reply:
minor: would be nice to apply linters > https://github.com/golangci/awesome-go-linters

Reply:
run gofmt